core/modules/parser/QSMySqlListener.cc

```diff
@@ -1441,15 +1475,10 @@ shared_ptr<query::SelectStmt> QSMySqlListener::getSelectStatement() const {
 template<typename ParentCBH, typename ChildAdapter, typename Context>
 shared_ptr<ChildAdapter> QSMySqlListener::pushAdapterStack(Context * ctx) {
     auto p = dynamic_pointer_cast<ParentCBH>(_adapterStack.top());
```

Comment:
Normally I would always put 'const' for variables which are not meant to be modified later:
auto const p = ...

Reply:
👍

Reply:
oh, but in this case `p` is passed into the ctor of the ChildAdapter, it becomes the child's parent, and receives callbacks which may change state; so it is not supposed to be const here.

Reply:
Adding const to the variable means that no one else can change that variable to point to a different object. The proposed earlier 'const'-ness is not 'viral' (it won't propagate down to an object referenced by the smart pointer). For the later you would need to:
auto p = dynamic_pointer_cast<ParentCBH const>
which I didn't suggest :-)